jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -334,7 +315,8 @@ private Path resolveTargetPath()
         {
             LOG.warn("bad alias ({} {}) for {}", e.getClass().getName(), e.getMessage(), path);
         }
-        return null;
+
+        return path.normalize();
```

Comment:
If there was an exception, we want to return `null`, as we could not resolve the target path.

Reply:
See I have changed the javadoc in `Resource.getTargetUri()` and changed the default implementation to `getUri()`.

So now the `getTargetUri()` is now only different to `getUri()` if there was an alias. Otherwise it will return the same value. And where `resolveTargetPath()` is used, we say it is an alias if the `targetPath` and `path` are different in any way.

So if path has unresolved dot segments we should resolve these here to show it is different to `path` (and should be an alias).